packages/components/addon/components/hds/app-footer/status-link.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+
+export const STATUSES = {
+  OPERATIONAL: {
+    text: 'Operational',
+    iconName: 'check-circle-fill',
+    iconColor: 'var(--token-color-foreground-success)',
+  },
+  DEGRADED: {
+    text: 'Degraded',
+    iconName: 'alert-triangle',
+    iconColor: 'var(--token-color-foreground-warning)',
+  },
+  OUTAGE: {
+    text: 'Outage',
+    iconName: 'x-square-fill',
+    iconColor: 'var(--token-color-foreground-critical)',
+  },
+  MAINTENANCE: {
+    text: 'Maintenance',
+    iconName: 'info-fill',
+    iconColor: 'var(--token-color-foreground-highlight)',
+  },
+};
+
+export default class HdsAppFooterStatusLinkComponent extends Component {
+  get status() {
+    return this.args.status.toUpperCase();
+  }
+
+  get statusIcon() {
+    return this.args.statusIcon ?? STATUSES[this.status.toUpperCase()].iconName;
+  }
+
+  get statusIconColor() {
+    return (
+      this.args.statusIconColor ?? STATUSES[this.status.toUpperCase()].iconColor
```

Comment:
what happens if as consumer
- I pass the `@statusIcon` argument
- I don't pass the `@statusIconColor` argument
- I don't pass the `@status` argument

Reply:
(this also applies to text, etc. consider different combinations: how do we solve this? should we instead require a `hash` object for argument?

Reply:
I pushed up a bunch of refactors. I changed it to require one of the preset statuses if the status argument is passed.

I don't know if this is what we would want or how else we'd want this to work.

Reply:
I believe I've resolved this. Please open another comment if you think I should handle things differently.